File: scratch/fix4.js
const fs = require('fs');
const file = 'src/app/(dashboard)/admin/pesquisas/page.tsx';
let content = fs.readFileSync(file, 'utf8');

const regex = /\/\/\s*ignore clicks on the action buttons area[\s\S]*?setIsModalOpen\(true\);/;
const replacement = `if ((e.target as HTMLElement).closest('button')) return;
    setEditandoId(p.id);
    setNovaPesquisa({
      nome: p.nome,
      tipo: p.tipo,
      induzida: p.induzida,
      multiSelect: p.multiSelect || false,
      metaDiaria: p.metaDiaria || 50,
      metaEntrevistas: p.metaEntrevistas || 1000,
      exigeMorador: p.exigeMorador || false,
      idadeMinima: p.idadeMinima || 16
    });
    setNovasOpcoes(p.opcoes.map(o => ({ nome: o.nome, partido: o.partido || '', foto: o.foto || '' })));
    setEntrevSelecionados([...p.entrevistadores]);
    setIsModalOpen(true);`;

content = content.replace(regex, replacement);
fs.writeFileSync(file, content, 'utf8');

File: fix4.py
with open('src/app/(dashboard)/admin/pesquisas/page.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

simulacao_old = """                            addPesquisa({
                              nome: `${pesquisaResultado.nome} — Simulação 2º Turno`,
                              tipo: 'Intenção de Voto',
                              status: 'Ativa',
                              induzida: true,
                              multiSelect: false,
                              metaDiaria: 50,
                              opcoes: [
                                { nome: c1.nome, partido: c1.partido, votos: 0 },
                                { nome: c2.nome, partido: c2.partido, votos: 0 },
                                { nome: 'Branco/Nulo', partido: '', votos: 0 },
                                { nome: 'Não Sabe', partido: '', votos: 0 }
                              ],
                              entrevistadores: pesquisaResultado.entrevistadores
                            });"""

simulacao_new = """                            addPesquisa({
                              nome: `${pesquisaResultado.nome} — Simulação 2º Turno`,
                              tipo: 'Intenção de Voto',
                              status: 'Ativa',
                              metaDiaria: 50,
                              perguntas: [
                                {
                                  id: `p${Date.now()}`,
                                  titulo: 'Se as eleições fossem hoje, em quem você votaria no 2º turno?',
                                  tipo: 'Induzida',
                                  multiSelect: false,
                                  opcoes: [
                                    { id: 'o1', nome: c1.nome, partido: c1.partido, votos: 0 },
                                    { id: 'o2', nome: c2.nome, partido: c2.partido, votos: 0 },
                                    { id: 'o3', nome: 'Branco/Nulo', partido: '', votos: 0 },
                                    { id: 'o4', nome: 'Não Sabe', partido: '', votos: 0 }
                                  ]
                                }
                              ],
                              entrevistadores: pesquisaResultado.entrevistadores
                            } as any);"""

content = content.replace(simulacao_old, simulacao_new)

with open('src/app/(dashboard)/admin/pesquisas/page.tsx', 'w', encoding='utf-8') as f:
    f.write(content)
